frontend/src/entities/configuration/configuration-reducer.ts

```diff
@@ -0,0 +1,130 @@
+import { defaultConfiguration, IAppConfiguration } from '../../shared/model/app.configuration.model';
+
+const initialState = {
+    appConfigList: [],
+    appConfig: defaultConfiguration,
+    failedToLoadConfig: false,
+};
+
+// Actions
+export const getConfiguration = (configScope: string) => {
+    return listConfigurations({ configScope, numVersions: 1});
+};
+
+export const listConfigurations = createAsyncThunk(
+    'app_config/list_configurations',
+    async (props: ListAppConfigurationProps) => {
+        const searchParams = new URLSearchParams();
+        searchParams.set('configScope', props.configScope);
+        if (props.numVersions) {
+            searchParams.set('numVersions', props.numVersions.toString());
+        }
+        const requestUrl = `/app-config?${searchParams}`;
+        
+        return await axios.get<IAppConfiguration[]>(requestUrl);
+    }
+);
+
+export const updateConfiguration = createAsyncThunk('app_config/update_app_config', 
+    async (newConfiguration: AppConfigurationProps, { rejectWithValue }) => {
+        const requestUrl = '/app-config';
+        try {
+            return await axios.post<IAppConfiguration>(requestUrl, newConfiguration.appConfiguration);
+        } catch (err) {
+            return rejectWithValue(err.response);
+        }
+    });
+
+// slice
+export const AppConfigSlice = createSlice({
+    name: 'appConfig',
+    initialState,
+    reducers: {
+        updateEntity (state, action: PayloadAction<IAppConfiguration>) {
+            state.appConfig = action.payload;
+        },
+    },
+    extraReducers (builder) {
+        builder
+            .addMatcher(isFulfilled(listConfigurations), (state, action: any) => {
+                const { data } = action.payload;
+                return {
+                    ...state,
+                    appConfig: data[0],
+                    appConfigList: data,
+                    failedToLoadConfig: false,
+                };
+            })
+            .addMatcher(isRejected(listConfigurations), (state) => {
+                console.log('Get rejected');
+                return {
+                    ...state,
+                    failedToLoadConfig: true,
+                };
+            })
+            .addMatcher(isPending(listConfigurations), (state) => {
+                return {
+                    ...state,
+                    failedToLoadConfig: false,
+                };
+            })
+            .addMatcher(isFulfilled(updateConfiguration), (state) => {
+                return {
+                    ...state,
+                };
+            })
+            .addMatcher(isRejected(updateConfiguration), (state) => {
+                return {
+                    ...state,
+                };
+            })
+            .addMatcher(isPending(updateConfiguration), (state) => {
+                return {
+                    ...state,
+                };
+            });
+            
+    },
+});
+
+// Interfaces
```

Comment:
nit: Types instead of interfaces? probably obvious anyways

Reply:
Good point, will rename